### ISSUE TITLE
Support typed strings on `no-unused-keys`

### ISSUE THREAD

**Issue by SimonSimCity** [intlify/eslint-plugin-vue-i18n#609]
**What rule do you want to change?**

`no-unused-keys`

**Does this change cause the rule to produce more or fewer warnings?**

fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**

Read the type for the input of the translation-function to determine the possible strings.

**Please provide some example code that this change will affect:**

```vue
<script lang="ts" setup>
let status: 'loading' | 'error' | 'success' = 'loading';
</script>

<template>
<span>{{ t(`global.${status}` }}</span>
</template>
```

**What does the rule currently do for this code?**

I guess it does ignore all template-strings, because I get a list of all these translation-keys reported as being unused.

**What will the rule do after it's changed?**

It recognizes, that the only available values (based on the given types) are `global.loading`, `global.error` and `global.success`, and will remember all those as being used. If not already, this could also be used in the rule `@intlify/vue-i18n/no-missing-keys` to determine possibly missing keys.

**Additional context**

* If TypeScript is not already used in this rule, it will most certainly reduce performance (https://github.com/intlify/eslint-plugin-vue-i18n/issues/354).
* Might have an intersection with https://github.com/intlify/eslint-plugin-vue-i18n/issues/602.
